ginga/web/pgw/Widgets.py

```diff
@@ -1918,6 +2465,66 @@ def add_widget(self, child, row, col, stretch=0):
                              value=self.render_body())
         self.make_callback('widget-added', child)
 
+    def insert_cell(self, row, col):
+        indices = [row, col]
+        if self._rendered:
+            app = self.get_app()
+            app.do_operation("insert_cell", id=self.id, value=indices)
+
+    def insert_row(self, index):
+        indices = [index, self.num_cols]
+        self.num_rows += 1
+
+        # handle case where user inserts row at the end of the gridbox
+        if index == self.num_rows - 1:
```

Comment:
Also do this if you decide to move the counter increment to the end of the method.

```suggestion
        if index == self.num_rows:
```